crates/cheatcodes/src/test/expect.rs

```diff
@@ -53,12 +53,27 @@ pub enum ExpectedCallType {
     Count,
 }
 
-#[derive(Clone, Debug, Default)]
+/// The type of expected revert.
+#[derive(Clone, Debug)]
+pub enum ExpectedRevertKind {
+    /// Expects revert from the next non-cheatcode call.
+    Default,
+    /// Expects revert from the next cheatcode call.
+    ///
+    /// The `pending` flag is used to track whether we have exited `expectCheatcodeRevert` context
+    /// or not. We have to track it to avoid expecting `expectCheatcodeRevert` call to revert
+    /// itself.
+    Cheatcode { pending: bool },
```

Comment:
I'd like a more expressive name for `pending` here.
perhaps simply `pending_processing` or even flip to `processed: bool`